### DISCUSSION TITLE
Change icon of type

### DISCUSSION THREAD

**Discussion by vlador4ik** [vakata/jstree#2662] [Q&A]
Hi everyone! I need change icon of type. Currently I init tree with some icons and after some data updates I need change icon for type folder and item for me.
I know about this:
`instance.set_icon(node)`
but I see that first parameter in this method should be specific node in tree but I need change icon for all nodes with specific type.

Can someone please help me? If you need additional information please write me.

**Comment by ApiumJacob** [comment]
Create a loop and iterate though your tree and change them one by one.  Here is how you get at you tree:

    var settings = $('#jstree_div').jstree(true).settings
    console.log(settings);
